tests/unit/library/test_json.py

```diff
@@ -1981,3 +1981,13 @@ def test_no_additionalProperties(self, compact):
             maybe_whitespace=True,
             compact=compact,
         )
+
+def test_ignored_keys_allowed_as_properties():
```

Comment:
@riedgar-ms is this test acceptable to you here? It doesn't *explicitly* test `id`; rather it checks against all ignored keys. Happy to specialize it to `id` if you like that better.

Reply:
Even better to check them all